### PR TITLE
Add position and size property signals

### PR DESCRIPTION
**Pull request by copycat-killer**
Add two new client properties: `property::size` and `property::position`.

Useful to avoid doubling signals, and to #1507 in particular.

### CODE REVIEW

**Comment by copycat-killer** [comment]
Seems that I caught latest 3 merge too. Is this a problem?

**Comment by psychon** [comment]
That should be fixable by rebasing this on master.
👍 
When someone else 👍's this, I can try to rebase this on master to get rid of the double changes. Feel free to rebase and force-push yourself, if you want.

**Comment by copycat-killer** [comment]
The problem is that I indeed rebased, but somehow it fetch my previous `git pull` too. 

I'll leave it to you. Thanks. :)

**Comment by Elv13** [comment]
There's still some rebase issue (the commit is now --author to actionless). Also, the doc is missing

```
--- When the client size changed.
-- @signal property::size
-- @see client.geometry


--- When the client position changed.
-- @signal property::position
-- @see client.geometry
```

**Comment by Elv13** [comment]
If you want to take a look, here's my rebase script https://gist.github.com/Elv13/eece84e72f815a231a746fca64fa339f (use `git rebase -i origin/master` after setting it as your git editor)

pick8

It's a small part of my larger "anti-IDE" workflow, but it is useful on its own.

**Comment by copycat-killer** [comment]
Added comments doc, but still unable to remove the intruder commit.

File: tests/test-awesomerc.lua [hunk 1]
```diff
@@ -42,22 +36,20 @@ local steps = {
...
-        -- local cb = get_callback({modkey}, " ")
-        -- assert(cb)
 
         --TODO use the key once the bug is fixed
         local l = old_c.screen.selected_tag.layout
         assert(l)
 
-        -- cb()
+        --awful.key.execute({modkey}, " ")
         awful.layout.inc(1)
```
Comment (Elv13) [inline tests/test-awesomerc.lua]: Maybe you forget to

    git fetch origin
    git rebase -i origin/master

Yes, git can be a git ;)

**Comment by Elv13** [comment]
Also, --author is still actionless. You can use `git --amand --author 'Luke Bonham <your@email>' --no-edit` (change your@email)

**Comment by copycat-killer** [comment]
I tried several things, but I fear I can't do anything now. 😠 

See if you can solve this, otherwise I'll do another PR again.

**Comment by Elv13** [comment]
hard one. Git has many ways to solve that, I used:

```
git cancel # an alias to reset HEAD --soft
git reset ../test-awesomerc*
git checkout ../test-awesomerc*
git commit -c 3d755951c060db528aaacdfe11ad003a3e297e08 -a
```

**Pull request closed without merging**